### ISSUE TITLE
deepvac_clothes_accept600 SOTA

### ISSUE THREAD

**Issue by gemfield** [DeepVAC/ESPNet#21]
# IoU 金榜
- test5: 58.6% (62.6%)
- test6: 58.4% (62.7%)
- test4: 57.8% (62.4%)
- test3: 57.4% (61.5%)
- test2: 57.2% (61.5%)
- test1: 53.6% (57.1%)

**Comment by gemfield** [comment]
# test占位符模版

## IoU
- label: 10%
- label_non_mix: 11%

## 特别超参
- 训练集：一个人像分割训练集
- config.input_w = 384
- config.input_h = 384

## 基本超参
- config.core.cls_num = 4
- config.aug.ImageWithMasksRandomRotateAug.max_angle = 45
- config.core.batch_size = 8
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 5e-3, momentum=0.9)
- config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[20,40,55,70,80,90,100,110,120,130,140,150,160], gamma=0.27030)
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- AugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 =>UDmotionAug@0.1 => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.15 => BrightnessJitterAug@0.15 =>ContrastJitterAug@0.15 => ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksNormalizeAug =>ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>ImageWithMasksToTensorAug', deepvac_config)

**Comment by buptlihang** [comment]
## test1

### IOU
- images：0.574
- images_no_dress: 

### 特别超参
- 训练集：std2.1
- 网络：semantic+detail+fusion
- config.input_w = 384
- config.input_h = 384

### 基本超参
- config.core.cls_num = 4
- config.core.batch_size = 16
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 5e-3, momentum=0.9)
- config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[20,40,55,70,80,90,100,110,120,130,140,150,160], gamma=0.27030)
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- BorderTargetAugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 => UDmotionAug@0.1 \
            => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.25 => BrightnessJitterAug@0.25 => ContrastJitterAug@0.25 => \
            ImageWithMasksRandomRotateAug@0.6 =>  ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 => \
            ImageWithMasksNormalizeAug => BorderTargetAug => ImageWithTwoMasksToTensorAug', deepvac_config)

**Comment by buptlihang** [comment]
## test1

### model
model_esp_epoch_248_miou_0.7251.pth

### IOU
- images: 0.536
- images_no_dress: 0.571
- val data: 0.7251

### 特别超参
- 训练集：std2.1
- 网络：original ESPNetV2
- config.input_w = 384
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 5e-3, momentum=0.9)
config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[20,40,55,70,80,90,100,110,120,130,140,150,160], gamma=0.27030)

### 基本超参
- config.core.cls_num = 4
- config.core.batch_size = 16
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- BorderTargetAugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 => UDmotionAug@0.1
=> NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.25 => BrightnessJitterAug@0.25 => ContrastJitterAug@0.25 =>
ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>
ImageWithMasksNormalizeAug => BorderTargetAug => ImageWithTwoMasksToTensorAug', deepvac_config)

**Comment by buptlihang** [comment]
## test2

### model
gemfield_epoch_186.pth

### IOU
- images: 0.572
- images_no_dress: 0.615
- val data: 0.73?

### 特别超参
- 训练集：std2.1
- 网络：original ESPNetV2
- config.input_w = 384
- config.input_h = 384
- config.core.optimizer = torch.optim.Adam(config.core.net.parameters(), 5e-4, (0.9, 0.999))
lambda_lr = lambda epoch: round ((1 - epoch/config.core.epoch_num) ** 0.9, 8)
config.core.scheduler = optim.lr_scheduler.LambdaLR(config.core.optimizer, lr_lambda=lambda_lr)

### 基本超参
- config.core.cls_num = 4
- config.core.batch_size = 16
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- BorderTargetAugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 => UDmotionAug@0.1
=> NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.25 => BrightnessJitterAug@0.25 => ContrastJitterAug@0.25 =>
ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>
ImageWithMasksNormalizeAug => BorderTargetAug => ImageWithTwoMasksToTensorAug', deepvac_config)

**Comment by buptlihang** [comment]
## test3

### model
model_fusion_0.7428.pth

### IOU
- images：0.574
- images_no_dress: 0.615
- val data: 0.7428

### 特别超参
- 训练集：std2.1
- 网络：semantic+detail+fusion
- config.input_w = 384
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 5e-3, momentum=0.9)
- config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[20,40,55,70,80,90,100,110,120,130,140,150,160], gamma=0.27030)

### 基本超参
- config.core.cls_num = 4
- config.core.batch_size = 16
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- BorderTargetAugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 => UDmotionAug@0.1 \
            => NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.25 => BrightnessJitterAug@0.25 => ContrastJitterAug@0.25 => \
            ImageWithMasksRandomRotateAug@0.6 =>  ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 => \
            ImageWithMasksNormalizeAug => BorderTargetAug => ImageWithTwoMasksToTensorAug', deepvac_config)

**Comment by buptlihang** [comment]
## test4

### model
model_fusion_0.7473.pth

### IOU
- images: 0.578
- images_no_dress: 0.624
- val data: 0.7473

### 特别超参
- 训练集：std2.1
- 网络：semantic+detail+fusion
- config.input_w = 384
- config.input_h = 384
- config.core.optimizer = torch.optim.Adam(config.core.net.parameters(), 5e-4, (0.9, 0.999))
lambda_lr = lambda epoch: round ((1 - epoch/config.core.epoch_num) ** 0.9, 8)
config.core.scheduler = optim.lr_scheduler.LambdaLR(config.core.optimizer, lr_lambda=lambda_lr)

### 基本超参
- config.core.cls_num = 4
- config.core.batch_size = 16
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- BorderTargetAugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 => UDmotionAug@0.1
=> NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.25 => BrightnessJitterAug@0.25 => ContrastJitterAug@0.25 =>
ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>
ImageWithMasksNormalizeAug => BorderTargetAug => ImageWithTwoMasksToTensorAug', deepvac_config)

**Comment by buptlihang** [comment]
## test5

### model
model_synthesis_epoch_155_0.7394.pth

### IOU
- images: 0.586
- images_no_dress: 0.626
- val data: 0.7394

### 特别超参
- 训练集：std2.1+synthesis(2800+5700)
- 网络：original ESPNetV2
- config.input_w = 384
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 5e-3, momentum=0.9)
config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[20,40,55,70,80,90,100,110,120,130,140,150,160], gamma=0.27030)

### 基本超参
- config.core.cls_num = 4
- config.core.batch_size = 16
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- BorderTargetAugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 => UDmotionAug@0.1
=> NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.25 => BrightnessJitterAug@0.25 => ContrastJitterAug@0.25 =>
ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>
ImageWithMasksNormalizeAug => BorderTargetAug => ImageWithTwoMasksToTensorAug', deepvac_config)

**Comment by buptlihang** [comment]
## test6

### model
model_synthesis_bg_0.7405.pth

### IOU
- images: 0.584
- images_no_dress: 0.627
- val data: 0.7405

### 特别超参
- 训练集：std2.1+Synthesis+bg(2800+5700+2000)
- 网络：original ESPNetV2
- config.input_w = 384
- config.input_h = 384
- config.core.optimizer = torch.optim.SGD(config.core.net.parameters(), 5e-3, momentum=0.9)
config.core.scheduler = optim.lr_scheduler.MultiStepLR(config.core.optimizer, milestones=[20,40,55,70,80,90,100,110,120,130,140,150,160], gamma=0.27030)

### 基本超参
- config.core.cls_num = 4
- config.core.batch_size = 16
- config.core.mean = config.data['mean']
- config.core.std = config.data['std']
- config.core.model_path = "/opt/public/pretrain/ESPNetv2/imagenet/espnetv2_s_2.0.pth"
- config.core.criterion = torch.nn.CrossEntropyLoss(weight)
- BorderTargetAugFactory('SpeckleAug@0.1 => GaussianAug@0.1 => HorlineAug@0.1 => VerlineAug@0.1 => LRmotionAug@0.1 => UDmotionAug@0.1
=> NoisyAug@0.1 => DarkAug@0.1 => ColorJitterAug@0.25 => BrightnessJitterAug@0.25 => ContrastJitterAug@0.25 =>
ImageWithMasksRandomRotateAug@0.6 => ImageWithMasksCenterCropAug => ImageWithMasksScaleAug => ImageWithMasksHFlipAug@0.5 =>
ImageWithMasksNormalizeAug => BorderTargetAug => ImageWithTwoMasksToTensorAug', deepvac_config)
